design/gitops-cluster-tracers.md

```diff
@@ -0,0 +1,40 @@
+# GitOps Cluster Tracers
+
+KOTS (Admin Console) does not have the ability to detect the currently deployed application 
+version in a GitOps cluster workflow. This proposal adds tracers (annotations) to the
+application pods so that it can be possible to detect that in the future.
+
+## Goals
+
+- Add annotations to application pods to help detect deployments in GitOps clusters in the future
+
+## Non Goals
+
+- Detect the currently deployed application version in a GitOps cluster
+
+## Background
+
+When customers decide to deploy their application via a GitOps workflow, they will
+probably want to know which version of the app is currently deployed in the cluster,
+which is currently not possible.
+
+## High-Level Design
+
+Add annotations for both the application slug and sequence to all of the application pods via Kustomize.
+Those annotations will be applied to the downstream manifests before commiting the new changes.
+
+## Detailed Design
+
+When (re)writing the midstream in KOTS, the following will be injected into the kustomization.yaml file:
```

Comment:
midtream/kustomization.yaml or downstreams/this-cluster/kustomization.yaml?

Reply:
I was discussing this with @divolgin, and he suggested the midstream. I believe it makes more sense to have it in the midstream, since the annotation will hold the app version and not the app downstream version?

Reply:
since we currently always have 1 downstream, i think it won’t matter, but if we decide to support multiple downstreams in the future, it is preferable to have those annotations for all the downstreams